From 501ac17ca941d7b61da4b4e942557a775c2d07af Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Tue, 22 Apr 2025 10:48:10 +0800
Subject: [PATCH] mtk: mt76: mt7996: rework the init value for current tx duty
 cycle

The init value of current tx duty cycle is 0 because firmware will not
send the unsolicited event with current tx duty cycle every second. Add
the init value as 100 to corrent the inin tx duty cycle.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 mt7996/mcu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 39f1bf1a..f25a6b4e 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -6286,6 +6286,8 @@ int mt7996_mcu_set_thermal_throttling(struct mt7996_phy *phy, u8 state)
 			return ret;
 	}
 
+	phy->throttle_state = state;
+
 	return 0;
 }
 
-- 
2.45.2

